lib/config.service.ts

```diff
@@ -8,7 +8,7 @@ import {
 import { NoInferType } from './types';
 
 @Injectable()
-export class ConfigService {
+export class ConfigService<K = any> {
```

Comment:
`keyof any` will fallback to `string | symbol | number` instead of `string`

Reply:
Hmm good point, would `ConfigService<K = { [key: string]: any }>` be more appropriate than?

Reply:
Have you tried `<K = Record<string, any>>`?

Reply:
Oh good call. That would definitely work.

Reply:
Updated :)